packages/cf-component-box/src/Box.js

```diff
@@ -0,0 +1,141 @@
+import React, { PropTypes } from 'react';
+import { createComponent } from 'cf-style-container';
+import propertiesSpec from './propertiesSpec';
+
+const propertyNames = Object.keys(propertiesSpec);
+
+// Generates styles given the provided props.
+//
+// This is driven by the metadata provided in propertiesSpec
+// and produces a datastructure like this:
+//
+// {
+//   margin: props.margin || props.theme['margin']
+//   width: props.width || props.theme['width']
+//   ...
+// }
+//
+// See propertySpec.js for more detail
+//
+const styles = props => {
+  let styles = {};
+
+  const propertyNames = Object.keys(propertiesSpec);
+  propertyNames.forEach(name => {
```

Comment:
Do we really need a for loop? I'm a bit worried about performance. We should probably avoid stuff like this considering how often this component will be used.